### PR TITLE
Qt/GeckoCodeWidget: Add option to add/remove gecko codes from UI 

### PR DESCRIPTION
**Pull request by spycrab**
Adds the option to manually add/remove/edit Gecko Codes from UI, as requested by @JMC47

### CODE REVIEW

**Comment by spycrab** [comment]
@MayImilae

File: Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp [hunk 1]
```diff
@@ -136,6 +156,56 @@ void GeckoCodeWidget::OnItemChanged(QListWidgetItem* item)
...
+  if (ed.exec())
+  {
+    SaveCodes();
+    UpdateList();
+  }
+}
+
+void GeckoCodeWidget::RemoveCode()
+{
+  auto* item = m_code_list->currentItem();
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp]: This can be a const pointer

File: Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp [hunk 2]
```diff
@@ -136,6 +156,56 @@ void GeckoCodeWidget::OnItemChanged(QListWidgetItem* item)
...
+  {
+    m_gecko_codes.push_back(std::move(code));
+    SaveCodes();
+    UpdateList();
+  }
+}
+
+void GeckoCodeWidget::EditCode()
+{
+  auto* item = m_code_list->currentItem();
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp]: This can be a const pointer

File: Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp [hunk 3]
```diff
@@ -136,6 +156,56 @@ void GeckoCodeWidget::OnItemChanged(QListWidgetItem* item)
   SaveCodes();
 }
 
+void GeckoCodeWidget::AddCode()
+{
+  CheatCodeEditor ed;
```
Comment (lioncash) [inline Source/Core/DolphinQt2/Config/GeckoCodeWidget.cpp]: This can be moved closer to it's usage spot.

**Pull request merged**